cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -768,9 +780,16 @@ Slice::CsGenerator::outputStreamWriter(const TypePtr& type, const string& scope,
     {
         if (isMappedToReadOnlyMemory(seq))
         {
-            builtin = BuiltinPtr::dynamicCast(seq->type());
-            out << "ZeroC.Ice.OutputStream.IceWriterFrom" << builtinSuffixTable[builtin->kind()]
-                << (forNestedType ? "Array" : "Sequence");
+            if (EnumPtr::dynamicCast(seq->type()))
+            {
+                out << helperName(type, scope) << ".IceWriterFrom" << (forNestedType ? "Array" : "Sequence");
+            }
+            else
+            {
+                builtin = BuiltinPtr::dynamicCast(seq->type());
+                out << "ZeroC.Ice.OutputStream.IceWriterFrom" << builtinSuffixTable[builtin->kind()]
```

Comment:
I saw that you added a new `builtinSuffix` method. It would be good to replace these instances where we directly reference the table with your new method.

Reply:
For a future PR.